code/modules/mob/living/simple_animal/guardian/guardian_abilities.dm

```diff
@@ -0,0 +1,733 @@
+			if(iscarbon(stando.target))
+				var/mob/living/carbon/C = stando.target
+				C.adjustBruteLoss(-5)
+				C.adjustFireLoss(-5)
+				C.adjustOxyLoss(-5)
+				C.adjustToxLoss(-5)
+				var/obj/effect/overlay/temp/heal/H = new /obj/effect/overlay/temp/heal(get_turf(C))
+				if(stando.namedatum)
+					H.color = stando.namedatum.colour
+				if(C == user)
+					stando.update_health_hud()
+					stando.med_hud_set_health()
+					stando.med_hud_set_status()
+
+/datum/guardian_abilities/heal/handle_mode()
+	if(stando.loc == user)
+		if(toggle)
+			stando.a_intent = initial(stando.a_intent)
+			stando.speed = initial(stando.speed)
+			stando.damage_coeff = initial(stando.damage_coeff)
+			stando.melee_damage_lower = initial(stando.melee_damage_lower)
+			stando.melee_damage_upper = initial(stando.melee_damage_upper)
+			stando << "<span class='danger'><B>You switch to combat mode.</span></B>"
+			toggle = FALSE
+		else
+			stando.a_intent = INTENT_HELP
+			stando.speed = 1
+			stando.damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 1, CLONE = 1, STAMINA = 0, OXY = 1)
+			stando.melee_damage_lower = 0
+			stando.melee_damage_upper = 0
+			stando << "<span class='danger'><B>You switch to healing mode.</span></B>"
+			toggle = TRUE
+	else
+		stando << "<span class='danger'><B>You have to be recalled to toggle modes!</span></B>"
+
+/datum/guardian_abilities/heal/alt_ability_act(atom/movable/A)
+	if(!istype(A))
+		return
+	if(stando.loc == user)
+		stando << "<span class='danger'><B>You must be manifested to warp a target!</span></B>"
+		return
+	if(!beacon)
+		stando << "<span class='danger'><B>You need a beacon placed to warp things!</span></B>"
+		return
+	if(!stando.Adjacent(A))
+		stando << "<span class='danger'><B>You must be adjacent to your target!</span></B>"
+		return
+	if(A.anchored)
+		stando << "<span class='danger'><B>Your target cannot be anchored!</span></B>"
+		return
+
+	var/turf/T = get_turf(A)
+	if(beacon.z != T.z)
+		stando << "<span class='danger'><B>The beacon is too far away to warp to!</span></B>"
+		return
+
+	stando << "<span class='danger'><B>You begin to warp [A].</span></B>"
+	A.visible_message("<span class='danger'>[A] starts to glow faintly!</span>", \
+	"<span class='userdanger'>You start to faintly glow, and you feel strangely weightless!</span>")
+	stando.do_attack_animation(A, null, 1)
+
+	if(!do_mob(stando, A, 60)) //now start the channel
+		stando << "<span class='danger'><B>You need to hold still!</span></B>"
+		return
+
+	new /obj/effect/overlay/temp/guardian/phase/out(T)
+	if(isliving(A))
+		var/mob/living/L = A
+		L.flash_act()
+	A.visible_message("<span class='danger'>[A] disappears in a flash of light!</span>", \
+	"<span class='userdanger'>Your vision is obscured by a flash of light!</span>")
+	do_teleport(A, beacon, 0)
+	new /obj/effect/overlay/temp/guardian/phase(get_turf(A))
+
+//metal pinky
+
+/datum/guardian_abilities/dextrous/handle_stats()
+	stando.dextrous = 1
+	stando.environment_target_typecache = list(
```

Comment:
This is also a problem; this NEEDS to be done and on the stand before the stand's New() does ..(), and you can't guarantee that if you have to reapply handle_stats().